project-management/src/main/java/life/qbic/projectmanagement/domain/service/SampleDomainService.java

```diff
@@ -42,21 +48,37 @@ public SampleDomainService(SampleRepository sampleRepository) {
   public Result<Collection<Sample>, ResponseCode> registerSamples(Project project,
       Map<SampleCode, SampleRegistrationRequest> sampleCodesToRegistrationRequests) {
     Objects.requireNonNull(sampleCodesToRegistrationRequests);
+
+    List<DomainEvent> domainEventsCache = new ArrayList<>();
+    var localDomainEventDispatcher = LocalDomainEventDispatcher.instance();
+    localDomainEventDispatcher.reset();
+    localDomainEventDispatcher.subscribe(
+        new SampleCreatedDomainEventSubscriber(domainEventsCache));
+
     Collection<Sample> samplesToRegister = new ArrayList<>();
     sampleCodesToRegistrationRequests.forEach((sampleCode, sampleRegistrationRequest) -> {
       var sample = Sample.create(sampleCode, sampleRegistrationRequest);
       samplesToRegister.add(sample);
     });
     Result<Collection<Sample>, ResponseCode> result = this.sampleRepository.addAll(project,
         samplesToRegister);
-    result.onValue(
-            createdSamples -> createdSamples.forEach(this::dispatchSuccessfulSampleRegistration))
-        .onError(Result::fromError);
+    if(result.isValue()) {
```

Comment:
because onValue expects me to do something with the sample (result) objects, which are not needed here at all. I don't have strong feelings about this, though, shall I change it anyway?